<!doctype html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>Using JBrowse Linear Genome View</title>

    <script
      src="https://unpkg.com/@jbrowse/react-linear-genome-view2/dist/react-linear-genome-view.umd.production.min.js"
      crossorigin
    ></script>
  </head>

  <body>
    <h2>Using JBrowse Linear Genome View!</h2>
    <button data-type="gene_button" data-location="10:94762681..94855547">
      CYP2C19
    </button>
    <button data-type="gene_button" data-location="13:32315086..32400266">
      BRCA2
    </button>
    <div id="jbrowse_linear_genome_view"></div>
    <p>
      The code for this app is available at
      <a
        href="https://github.com/GMOD/jbrowse-react-linear-genome-view-vanillajs-demo"
      >
        https://github.com/GMOD/jbrowse-react-linear-genome-view-vanillajs-demo
      </a>
    </p>

    <button id="showviewstate">Show view state</button>
    <textarea id="viewstate" name="viewstate" rows="20" cols="80"></textarea>
    <p>updates:</p>
    <textarea
      id="update"
      name="update"
      rows="5"
      cols="80"
      wrap="off"
    ></textarea>
    <script>
      const { React, createRoot, createViewState, JBrowseLinearGenomeView } =
        JBrowseReactLinearGenomeView;

      const updates = document.getElementById('update');

      const state = new createViewState({
        assembly: {
          name: 'hg38',
          sequence: {
            type: 'ReferenceSequenceTrack',
            trackId: 'GRCh38-ReferenceSequenceTrack',
            adapter: {
              type: 'BgzipFastaAdapter',
              uri: 'https://jbrowse.org/genomes/GRCh38/fasta/hg38.prefix.fa.gz',
            },
          },
          refNameAliases: {
            adapter: {
              type: 'RefNameAliasAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/hg38_aliases.txt',
            },
          },
          cytobands: {
            adapter: {
              type: 'CytobandAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/cytoBand.txt',
            },
          },
        },
        tracks: [
          {
            type: 'FeatureTrack',
            trackId: 'genes',
            name: 'NCBI RefSeq Genes',
            assemblyNames: ['hg38'],
            category: ['Genes'],
            adapter: {
              type: 'Gff3TabixAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/ncbi_refseq/GCA_000001405.15_GRCh38_full_analysis_set.refseq_annotation.sorted.gff.gz',
            },
            textSearching: {
              textSearchAdapter: {
                type: 'TrixTextSearchAdapter',
                textSearchAdapterId: 'gff3tabix_genes-index',
                uri: 'https://jbrowse.org/genomes/GRCh38/ncbi_refseq/trix/GCA_000001405.15_GRCh38_full_analysis_set.refseq_annotation.sorted.gff.gz.ix',
                assemblyNames: ['GRCh38'],
              },
            },
          },
          {
            type: 'FeatureTrack',
            trackId: 'repeats_hg38',
            name: 'Repeats',
            assemblyNames: ['hg38'],
            category: ['Annotation'],
            adapter: {
              type: 'BigBedAdapter',
              uri: 'https://jbrowse.org/genomes/GRCh38/repeats.bb',
            },
          },
          {
            type: 'AlignmentsTrack',
            trackId: 'NA12878.alt_bwamem_GRCh38DH.20150826.CEU.exome',
            name: 'NA12878 Exome',
            assemblyNames: ['hg38'],
            category: ['1000 Genomes', 'Alignments'],
            adapter: {
              type: 'CramAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/alignments/NA12878/NA12878.alt_bwamem_GRCh38DH.20150826.CEU.exome.cram',

              sequenceAdapter: {
                type: 'BgzipFastaAdapter',
                uri: 'https://jbrowse.org/genomes/GRCh38/fasta/hg38.prefix.fa.gz',
              },
            },
          },
          {
            type: 'VariantTrack',
            trackId:
              'ALL.wgs.shapeit2_integrated_snvindels_v2a.GRCh38.27022019.sites.vcf',
            name: '1000 Genomes Variant Calls',
            assemblyNames: ['hg38'],
            category: ['1000 Genomes', 'Variants'],
            adapter: {
              type: 'VcfTabixAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/variants/ALL.wgs.shapeit2_integrated_snvindels_v2a.GRCh38.27022019.sites.vcf.gz',
            },
          },
          {
            type: 'QuantitativeTrack',
            trackId: 'hg38.100way.phyloP100way',
            name: 'hg38.100way.phyloP100way',
            category: ['Conservation'],
            assemblyNames: ['hg38'],
            adapter: {
              type: 'BigWigAdapter',
              uri: 'https://hgdownload.cse.ucsc.edu/goldenpath/hg38/phyloP100way/hg38.phyloP100way.bw',
            },
          },
          {
            type: 'AlignmentsTrack',
            trackId: 'skbr3_pacbio',
            name: 'SKBR3 pacbio',
            assemblyNames: ['hg38'],
            adapter: {
              type: 'BamAdapter',
              uri: 'https://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/skbr3/SKBR3_Feb17_GRCh38.sorted.bam',
            },
          },
        ],
        defaultSession: {
          name: 'this session',
          margin: 0,
          view: {
            id: 'linearGenomeView',
            type: 'LinearGenomeView',
            init: {
              assembly: 'hg38',
              loc: '10:29,838,565..29,838,850',
              tracks: [
                'GRCh38-ReferenceSequenceTrack',
                'genes',
                'NA12878.alt_bwamem_GRCh38DH.20150826.CEU.exome',
                'hg38.100way.phyloP100way',
                'ALL.wgs.shapeit2_integrated_snvindels_v2a.GRCh38.27022019.sites.vcf',
              ],
            },
          },
        },
        onChange: patch => {
          updates.innerHTML += JSON.stringify(patch) + '\n';
        },
      });

      function navTo(event) {
        state.session.view.navToLocString(event.target.dataset.location);
      }
      const buttons = document.getElementsByTagName('button');
      for (const button of buttons) {
        if (button.dataset.type === 'gene_button') {
          button.addEventListener('click', navTo);
        }
      }

      const textArea = document.getElementById('viewstate');
      document.getElementById('showviewstate').addEventListener('click', () => {
        textArea.innerHTML = JSON.stringify(state.session.view, undefined, 2);
      });

      const root = createRoot(
        document.getElementById('jbrowse_linear_genome_view'),
      );
      root.render(
        React.createElement(JBrowseLinearGenomeView, {
          viewState: state,
        }),
      );
    </script>
  </body>
</html>
